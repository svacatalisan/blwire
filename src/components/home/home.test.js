import React from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import Home from './home';
import ReactTestUtils from 'react-dom/test-utils';

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

describe('Home Component', () => {
  let store;
  let component;
  let container;
  
  beforeEach(() => {
    store = mockStore({
      data: {
        shapes: [
          {
              "id": 1,
              "color": "purple",
              "shape": "round"
          },
          {
              "id": 2,
              "color": "red",
              "shape": "round"
          }],
        filters: [
          {
              "id": 1,
              "type": "literal",
              "text": "Round"
          },
          {
              "id": 2,
              "type": "literal",
              "text": "Square"
          }],
          appliedFilters: [
            {
                "id": 1,
                "type": "literal",
                "text": "Round"
            },
            {
                "id": 2,
                "type": "literal",
                "text": "Square"
            }] 
      },
    });

    store.dispatch = jest.fn();

    component = create(
      <Provider store={store}>
        <Home />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {
    container = ReactTestUtils.renderIntoDocument(
      <Provider store={store}>
        <Home />
    </Provider>);
    
    console.log(container);
  });
});
