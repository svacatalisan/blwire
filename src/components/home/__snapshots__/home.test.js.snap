// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render with given state from Redux store 1`] = `
<div
  className="home"
>
  <div
    className="filters"
  >
    <div
      className="literal-filters"
    >
      <div
        className="filter-item"
        onClick={[Function]}
      >
        <span
          style={
            Object {
              "color": "black",
            }
          }
        >
          Round
        </span>
      </div>
      <div
        className="filter-item"
        onClick={[Function]}
      >
        <span
          style={
            Object {
              "color": "black",
            }
          }
        >
          Square
        </span>
      </div>
    </div>
    <div
      className="color-filters"
    />
  </div>
  <div
    className="items-number-container"
  >
    <div
      className="items-number"
    >
      <div
        className="title"
      >
        At least one criterion should be selected from each filter categories
      </div>
    </div>
  </div>
  <div
    className="shapes"
  >
    <span
      className="hide"
    />
    <span
      className="hide"
    />
    <div
      className="inline-block"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <div
        className="shape"
      >
        <div
          style={
            Object {
              "background": undefined,
            }
          }
        />
      </div>
    </div>
    <div
      className="inline-block"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <div
        className="shape"
      >
        <div
          style={
            Object {
              "background": undefined,
            }
          }
        />
      </div>
    </div>
    <div
      className="inline-block"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <div
        className="shape"
      >
        <div
          style={
            Object {
              "background": undefined,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
